### PR TITLE
getting started docs improvements

### PR DESCRIPTION
**Pull request by choldgraf**
Adds contextual information to the deployment guide / travis

### CODE REVIEW

**Comment by choldgraf** [comment]
@mpacer or @yuvipanda wanna take a quick glance at this? it's the doc notes/changes from our meeting yesterday...ready to go from my perspective

File: docs/source/deployment/how.md [hunk 1]
```diff
@@ -37,9 +38,9 @@ which we step through below.
    .. image:: travis-screenshot.png
    ```
 
-4. In your fork of the [mybinder.org-deploy][https://github.com/jupyterhub/mybinder.org-deploy]
+6. In your fork of the [mybinder.org-deploy](https://github.com/jupyterhub/mybinder.org-deploy)
    repository, open `mybinder/requirements.yaml`.
-5. Toward the end of the file, you will see lines similar to:
+7. Toward the end of the file, you will see lines similar to:
```
Comment (betatim) [inline docs/source/deployment/how.md]: Should we update this advice to point to https://github.com/jupyterhub/mybinder.org-deploy/#key-links (or links therein). This is what I actually use as it is far easier to find what I am looking for.

**Comment by betatim** [comment]
LGTM modulo the comment.

File: docs/source/deployment/how.md [hunk 2]
```diff
@@ -23,23 +23,21 @@ that is used in mybinder.org corresponds to updating the BinderHub helm chart ve
...
 
 1. Merge changes to BinderHub.
-2. Open the [Travis build for BinderHub](https://travis-ci.org/jupyterhub/binderhub),
-   navigate to the page corresponding to the master branch, `TEST=helm` job.
-3. If the build succeeds, grab the hash that is displayed at the end of the
-   travis output. It looks something like:
+2. Open the [branches page for the BinderHub travis account](https://travis-ci.org/jupyterhub/binderhub/branches).
+3. Click on the number for the latest build on "Master". It will say "#NNN passed".
+4. You'll see a list of builds that have been run on this branch. Click on the `TEST=helm` job.
+5. If the build succeeds (is green), a new helm chart for BinderHub will automatically
```
Comment (mpacer) [inline docs/source/deployment/how.md]: @choldgraf @yuvipanda Did we do this the other day? I don't remember this step…

Comment (choldgraf) [reply docs/source/deployment/how.md]: I believe we bumped repo2docker, not binderhub, so nope we didn't cover this

**Comment by betatim** [comment]
Merged. We can keep fine tuning in a new PR.

**Pull request merged**